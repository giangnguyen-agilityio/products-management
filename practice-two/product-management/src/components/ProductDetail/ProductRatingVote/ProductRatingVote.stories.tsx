import type { Meta, StoryObj } from '@storybook/react'
import ProductRatingVote from './index'
const meta = {
  title: 'Practice Two/ProductRatingVote Component',
  component: ProductRatingVote,
  decorators: [
    Story => (
      <div
        style={{
          padding: '10px',
        }}
      >
        <Story />
      </div>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof ProductRatingVote>
export default meta
type Story = StoryObj<typeof meta>
export const Primary: Story = {
  args: {
    ratingVote: 5,
  },
}
export const Secondary: Story = {
  args: {
    ratingVote: 3.5,
  },
}
