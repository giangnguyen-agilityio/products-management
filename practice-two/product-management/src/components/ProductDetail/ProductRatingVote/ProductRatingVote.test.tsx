import { render } from '@testing-library/react'
import ProductRatingVote from './'

jest.mock('@assets/icons/empty_star_icon.svg', () => ({
  default: 'emptyStar',
}))

jest.mock('@assets/icons/half_star_icon.svg', () => ({
  default: 'halfStar',
}))

jest.mock('@assets/icons/half_star_icon.svg', () => ({
  default: 'fullStar',
}))

describe('ProductRatingVote Component', () => {
  it('renders correct number of stars for a given rating', () => {
    const ratingVote = 3.5
    const { container } = render(<ProductRatingVote ratingVote={ratingVote} />)

    expect(container).toMatchSnapshot()
  })
})
