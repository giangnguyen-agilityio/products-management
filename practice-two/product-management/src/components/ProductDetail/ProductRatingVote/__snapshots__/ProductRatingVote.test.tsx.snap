// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRatingVote Component renders correct number of stars for a given rating 1`] = `
<div>
  <div
    class="rating-vote css-losko6"
  >
    <div
      aria-label="Full star"
      class="star css-0"
    >
      <img
        alt="The full star icon"
        class="chakra-image css-1fjwirh"
        src=""
      />
    </div>
    <div
      aria-label="Full star"
      class="star css-0"
    >
      <img
        alt="The full star icon"
        class="chakra-image css-1fjwirh"
        src=""
      />
    </div>
    <div
      aria-label="Full star"
      class="star css-0"
    >
      <img
        alt="The full star icon"
        class="chakra-image css-1fjwirh"
        src=""
      />
    </div>
    <div
      aria-label="Half star"
      class="star css-0"
    >
      <img
        alt="The half star icon"
        class="chakra-image css-1fjwirh"
        src="[object Object]"
      />
    </div>
    <div
      aria-label="Empty star"
      class="star css-0"
    >
      <img
        alt="The empty star icon"
        class="chakra-image css-1fjwirh"
        src="[object Object]"
      />
    </div>
  </div>
</div>
`;
